### ISSUE TITLE
AttributeError: 'numpy.bool_' object has no attribute 'startswith'

### ISSUE THREAD

**Issue by sarath-mec** [mwaskom/seaborn#3505]
Getting the following error while running the code, when specifying hue as a boolean type, as follows. In the below code block 
`alone` is a `bool` type

### **Code**

```
import pandas as pd
import seaborn as sns
df = sns.load_dataset('titanic')
df.info()

sns.set(style="whitegrid", color_codes=True)
sns.countplot(x="sex", hue= "alone", data=df)
```
### **Error**
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Untitled-1.ipynb Cell 1 line 7
      [4](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=3) df.info()
      [6](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=5) sns.set(style="whitegrid", color_codes=True)
----> [7](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=6) sns.countplot(x="sex", hue= "alone", data=df)

File [c:\Users\xxxx\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\categorical.py:2955](file:///C:/Users/xxxx/AppData/Local/anaconda3/envs/ai_ml_training/Lib/site-packages/seaborn/categorical.py:2955), in countplot(data, x, y, hue, order, hue_order, orient, color, palette, saturation, width, dodge, ax, **kwargs)
   2952 if ax is None:
   2953     ax = plt.gca()
-> 2955 plotter.plot(ax, kwargs)
   2956 return ax

File [c:\Users\xxxx\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\categorical.py:1587](file:///C:/Users/xxxx/AppData/Local/anaconda3/envs/ai_ml_training/Lib/site-packages/seaborn/categorical.py:1587), in _BarPlotter.plot(self, ax, bar_kws)
   1585 """Make the plot."""
   1586 self.draw_bars(ax, bar_kws)
-> 1587 self.annotate_axes(ax)
   1588 if self.orient == "h":
   1589     ax.invert_yaxis()

File [c:\Users\xxxx\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\categorical.py:767](file:///C:/Users/xxxx/AppData/Local/anaconda3/envs/ai_ml_training/Lib/site-packages/seaborn/categorical.py:767), in _CategoricalPlotter.annotate_axes(self, ax)
    764     ax.set_ylim(-.5, len(self.plot_data) - .5, auto=None)
    766 if self.hue_names is not None:
--> 767     ax.legend(loc="best", title=self.hue_title)
...
-> 1291     if label and not label.startswith('_'):
   1292         handles.append(handle)
   1293         labels.append(label)
```

The error is occurring in the following code
[/matplotlib/blob/main/lib/matplotlib/legend.py#L1283](https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/legend.py#L1283)

Should we have something similar to 
[matplotlib/blob/main/lib/matplotlib/legend.py#L460](https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/legend.py#L460)

**Comment by thuiop** [comment]
I can replicate with an older version but this seems to have been fixed on master ; I guess the support for boolean categories was broken at some point. I suggest updating to the current release candidate or waiting for the official 0.13 release.

**Comment by mwaskom** [comment]
FWIW I actually cannot replicate on 0.12.2 with matplotlib 3.7.0, numpy 1.24.2, and pandas 2.0.0, so could also be an upstream change that seaborn has since adapted to.

**Comment by sarath-mec** [comment]
I have 

matplotlib=>3.8.0
numpy=>1.26.0
pandas=>2.1.1
seaborn=>0.12.2

```
(ai_ml_training) PS C:\Users\sarsasid> pip install matplotlib --upgrade
Requirement already satisfied: matplotlib in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (3.8.0)
Requirement already satisfied: contourpy>=1.0.1 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (1.1.1)
Requirement already satisfied: cycler>=0.10 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (4.42.1)
Requirement already satisfied: kiwisolver>=1.0.1 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (1.4.5)
Requirement already satisfied: numpy<2,>=1.21 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from matplotlib) (1.26.0)
Requirement already satisfied: packaging>=20.0 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (23.1)
Requirement already satisfied: pillow>=6.2.0 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (10.0.1)
Requirement already satisfied: pyparsing>=2.3.1 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (3.1.1)
Requirement already satisfied: python-dateutil>=2.7 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib) (2.8.2)
Requirement already satisfied: six>=1.5 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from python-dateutil>=2.7->matplotlib) (1.16.0)
(ai_ml_training) PS C:\Users\sarsasid> pip install numpy --upgrade
Requirement already satisfied: numpy in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (1.26.0)
(ai_ml_training) PS C:\Users\sarsasid> pip install pandas --upgrade
Requirement already satisfied: pandas in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (2.1.1)
Requirement already satisfied: numpy>=1.23.2 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from pandas) (1.26.0)
Requirement already satisfied: python-dateutil>=2.8.2 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from pandas) (2023.3)
Requirement already satisfied: six>=1.5 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
(ai_ml_training) PS C:\Users\sarsasid> pip install matplotlib_inline --upgrade
Requirement already satisfied: matplotlib_inline in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (0.1.6)
Requirement already satisfied: traitlets in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib_inline) (5.9.0)
(ai_ml_training) PS C:\Users\sarsasid> pip install seaborn --upgrade
Requirement already satisfied: seaborn in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (0.12.2)
Requirement already satisfied: numpy!=1.24.0,>=1.17 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from seaborn) (1.26.0)
Requirement already satisfied: pandas>=0.25 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from seaborn) (2.1.1)
Requirement already satisfied: matplotlib!=3.6.1,>=3.1 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from seaborn) (3.8.0)
Requirement already satisfied: contourpy>=1.0.1 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (1.1.1)
Requirement already satisfied: cycler>=0.10 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (4.42.1)
Requirement already satisfied: kiwisolver>=1.0.1 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (1.4.5)
Requirement already satisfied: packaging>=20.0 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (23.1)
Requirement already satisfied: pillow>=6.2.0 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (10.0.1)
Requirement already satisfied: pyparsing>=2.3.1 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (3.1.1)
Requirement already satisfied: python-dateutil>=2.7 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from matplotlib!=3.6.1,>=3.1->seaborn) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from pandas>=0.25->seaborn) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in c:\users\sarsasid\appdata\local\anaconda3\envs\ai_ml_training\lib\site-packages (from pandas>=0.25->seaborn) (2023.3)
Requirement already satisfied: six>=1.5 in c:\users\sarsasid\appdata\roaming\python\python311\site-packages (from python-dateutil>=2.7->matplotlib!=3.6.1,>=3.1->seaborn) (1.16.0)
(ai_ml_training) PS C:\Users\sarsasid> python
Python 3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import seaborn as sns
>>> df = sns.load_dataset('titanic')
>>> df.info()
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   survived     891 non-null    int64
 1   pclass       891 non-null    int64
 2   sex          891 non-null    object
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64
 5   parch        891 non-null    int64
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object
 8   class        891 non-null    category
 9   who          891 non-null    object
 10  adult_male   891 non-null    bool
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object
 13  alive        891 non-null    object
 14  alone        891 non-null    bool
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB
>>>
>>> sns.set(style="whitegrid", color_codes=True)
>>> sns.countplot(x="sex", hue= "alone", data=df)
C:\Users\sarsasid\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\_oldcore.py:1498: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead
  if pd.api.types.is_categorical_dtype(vector):
C:\Users\sarsasid\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\_oldcore.py:1498: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead
  if pd.api.types.is_categorical_dtype(vector):
C:\Users\sarsasid\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\_oldcore.py:1498: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead
  if pd.api.types.is_categorical_dtype(vector):
C:\Users\sarsasid\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\_oldcore.py:1498: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead
  if pd.api.types.is_categorical_dtype(vector):
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\sarsasid\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\categorical.py", line 2955, in countplot
    plotter.plot(ax, kwargs)
  File "C:\Users\sarsasid\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\categorical.py", line 1587, in plot
    self.annotate_axes(ax)
  File "C:\Users\sarsasid\AppData\Local\anaconda3\envs\ai_ml_training\Lib\site-packages\seaborn\categorical.py", line 767, in annotate_axes
    ax.legend(loc="best", title=self.hue_title)
  File "C:\Users\sarsasid\AppData\Roaming\Python\Python311\site-packages\matplotlib\axes\_axes.py", line 322, in legend
    handles, labels, kwargs = mlegend._parse_legend_args([self], *args, **kwargs)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\sarsasid\AppData\Roaming\Python\Python311\site-packages\matplotlib\legend.py", line 1361, in _parse_legend_args
    handles, labels = _get_legend_handles_labels(axs, handlers)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\sarsasid\AppData\Roaming\Python\Python311\site-packages\matplotlib\legend.py", line 1291, in _get_legend_handles_labels
    if label and not label.startswith('_'):
                     ^^^^^^^^^^^^^^^^
AttributeError: 'numpy.bool_' object has no attribute 'startswith'
>>>
```

**Comment by mwaskom** [comment]
As @thuiop said, this is replicable on latest versions but fixed on the master branch of seaborn and will be in the forthcoming v0.13.0 release. You can update to the v0.13.0 release candidate to try if you want.

**Comment by sarath-mec** [comment]
Yes, Thanks This issue seems to be fixed in v0.13.0rc0. I did not notice the comment

https://github.com/mwaskom/seaborn/releases/tag/v0.13.0rc0

Fix is v
`pip install seaborn~=0.13.0rc0`
